Generalize and reorganize social media links #39

domingohui opened the pull request:
Based on #39 
Will be adding svg's for the remaining links soon :)

ashmaroli commented on _includes/footer.html:
```diff
@@ -22,22 +22,25 @@ <h2 class="footer-heading">{{ site.title | escape }}</h2>
 
       <div class="footer-col footer-col-3">
-        <p>{{ site.description | escape }}</p>
+          <p>{{ site.description | escape }}</p>
```
this change is unnecessary IMO

domingohui replied:
oops this just came from my auto indentation..

ashmaroli commented on _sass/minima/_base.scss:
```diff
@@ -188,11 +188,21 @@ pre {
+    display: inline-block;
+
+    &.github {
+        background-image: url("data:image/svg+xml;charset=utf-8,<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 16 16'><path fill='#828282' d='M7.999,0.431c-4.285,0-7.76,3.474-7.76,7.761 c0,3.428,2.223,6.337,5.307,7.363c0.388,0.071,0.53-0.168,0.53-0.374c0-0.184-0.007-0.672-0.01-1.32 c-2.159,0.469-2.614-1.04-2.614-1.04c-0.353-0.896-0.862-1.135-0.862-1.135c-0.705-0.481,0.053-0.472,0.053-0.472  ...
```
I recommend using minima's `$grey-color` variable here instead of the hex-code for `fill`

ashmaroli commented on example/about.md:
```diff
@@ -7,9 +7,9 @@ permalink: /about/
 You can find the source code for the Jekyll new theme at:
-{% include icon-github.html username="jekyll" %} /
+<span class="svg-icon github"></span> /
 [minima](https://github.com/jekyll/minima)
```
You missed the link to `[jekyll-organisation]`. Now its just `[icon] / [minima-repo]`.
Same mistake below

domingohui commented:
Thanks @ashmaroli!  I will make the changes tonight.

domingohui commented:
Credits to @danleech. I'm using svg icons from [here](https://github.com/danleech/simple-icons/tree/gh-pages/icons).

But I will probably need to fix up the youtube icon later. @ashmaroli Please build it and let me know what you think thanks!

domingohui commented:
Are you referring to the way it was set up before? (e.g. [icon-twitter.html](https://github.com/jekyll/minima/blob/master/_includes/icon-twitter.html) and [icon-twitter.svg](https://github.com/jekyll/minima/blob/master/_includes/icon-twitter.svg))

That's a good idea. Having everything in a giant scss file is a bit awkward too. I'm thinking maybe we can just use a single html file including the svg, instead of having a separate svg file.

pathawks commented on _sass/minima/_base.scss:
```diff
@@ -188,11 +188,60 @@ pre {
+    display: inline-block;
+
+    &.github {
+        background-image: url("data:image/svg+xml;charset=utf-8,<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 16 16'><path fill='#{$grey-color}' d='M7.999,0.431c-4.285,0-7.76,3.474-7.76,7.761 c0,3.428,2.223,6.337,5.307,7.363c0.388,0.071,0.53-0.168,0.53-0.374c0-0.184-0.007-0.672-0.01-1.32 c-2.159,0.469-2.614-1.04-2.614-1.04c-0.353-0.896-0.862-1.135-0.862-1.135c-0.705-0.481,0.053-0.472,0.053 ...
```
I am referring to this: inlining the SVG data inside the CSS file, instead of using external files.

domingohui replied:
But you were saying it would be better if each svg data was an external file?

pathawks replied:
Yes, perhaps it would be better to put the SVGs each in separate files.

ashmaroli commented on example/about.md:
```diff
@@ -7,9 +7,9 @@ permalink: /about/
 You can find the source code for Jekyll at
-<span class="svg-icon github"></span> /
+<a href="https://github.com/{{ site.github_username }}"><span class="svg-icon github"></span><span class="username">{{ site.github_username }}</span></a> /
 [jekyll](https://github.com/jekyll/jekyll)
```
`<a href="https://github.com/{{ site.github_username }}">[...]</a>` doesn't make sense here. It simply needs to point to the Jekyll organisation.

```
You can find the source code for the Jekyll new theme at:
<span class="svg-icon github"></span> [jekyll]((https://github.com/jekyll) /
[minima](https://github.com/jekyll/minima)
```
You may also take a cue from this [PR](https://github.com/jekyll/jekyll/pull/5903). The difference is I left out the icons entirely as part of the PR. I feel, its cleaner that way.

domingohui replied:
True. I must have been super tired... Thanks!

danleech commented:
@pathawks’ HTTP/2-based argument for not inlining is a strong one. If you *are* inlining, I’d recommend doing so in HTML, not CSS. That way you can style `path`, `circle`, `rect` etc with CSS.

For example (in Sass):

```css
.icon-link {
    color: blue;
    svg {
        path, circle, rect {
            fill: blue;
        }
    }
    &:focus, &:hover {
        color: red;
        svg {
            path, circle, rect {
                fill: red;
            }
        }
    }
}
```

caspervonb commented:
What about using `site.social.links` (which is an array of links, see jekyll-seo)

chriscannon commented:
This really should be merged. What's the hold up?

benbalter commented:
@domingohui Can you provide some screenshots of what's going on here and why? I'm a bit weary that Jekyll's default starter theme should have includes with non-human-readable and thus black-block SVGs as it doesn't go a long way to help new users understand the basics of Jekyll (where as an include with HTML is a bit more transparent).

DirtyF commented:
I would avoid includes here and just group all SVG in a single file and use `svg` tags as described here:
https://fvsch.com/code/svg-icons/how-to/

domingohui commented:
hey @benbalter @DirtyF  thanks for the feedback. I don't have much experience working with SVG. Sorry if I made anything more confusing than necessary! But I guess there are some improvements I can do with this PR:

- Condense the link ```<a>``` part in [```{% if site.dribbble_username %}<li><a href="https://dribbble.com/{{ site.dribbble_username }}">{% include icons/dribbble.svg %}<span class="username">{{ site.dribbble_username }}</span></a></li>{% endif %}```](https://github.com/jekyll/minima/pull/112/files#diff-7693eb643e8fe8b8d46b407909b408ceR26) into a single ```dribbble.html```. 
So all we have to do is ```{% include dribbble.html%}```. I prefer to separate the HTML part and the svg part. So we would have a ```dribbble.svg``` and a ```dribbble.html``` that ```include```s the ```svg```, for example. What do you think? I think this is also the original setup...

- Remove the ```fill``` tag in svg's. I overlooked that as I just overrode it in css...

- Combine all svg's in one svg sprite. 

What do you think? I will push again soon.

domingohui commented:
@chriscannon are you referring to [jekyll-seo-tag](https://github.com/jekyll/jekyll-seo-tag)? I'd happy to add it too. It's a great idea. But I don't know if we want to use another plugin in this specific theme. @pathawks What do you think?

chriscannon commented:
@domingohui all my PR is doing is making sure you're is using the `site.linkedin_username` in the linkedin link instead of `site.github_username`. I assume this was just a simple copy/paste error.

domingohui commented:
@chriscannon Oops wrong tag... Sorry, but I will double check the links for sure.

DirtyF commented on _includes/social_links/dribbble.html:
```diff
@@ -0,0 +1 @@
+<li><a href="https://dribbble.com/{{ site.dribbble_username }}"><svg class="svg-icon" xmlns="http://www.w3.org/2000/svg"><use xlink:href="/assets/icons.svg#dribbble"></use></svg> <span class="username">{{ site.dribbble_username }}</span></a></li>
```
why add the `<li>` tag in the include file? What if you want to use the SVG outside of a list ?

domingohui replied:
Good call..

DirtyF commented on assets/icons.svg:
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="iso-8859-1"?>
```
You can get rid of the encoding declaration here.

DirtyF commented:
@benbalter @jglovier 

Here's a screenshot:

![minima-footer](https://cloud.githubusercontent.com/assets/103008/25042502/d58f7134-2118-11e7-91c8-c671e5da21f5.png)

domingohui commented:
Fixed the spacing. The spacing was only applied when the icons were in a social media list. Is this intended? i.e. The spacing wouldn't apply to the github icons in ```about.md```.

I also added the email icon in ```index.md```. Should we keep it?

ashmaroli commented:
> The spacing wouldn't apply to the github icons in about.md and other places.

The icon-spacing within "content" pages is now within the scope of this PR. Please adjust the CSS for this accordingly when you fix the vertical-alignment issue.

domingohui commented:
Also, @DirtyF could you pinpoint where the rest of vertical alignment issues would be? Thanks! 
![image](https://cloud.githubusercontent.com/assets/5943822/25068717/d1ab296e-223a-11e7-90cc-25de915d682e.png)

DirtyF commented:
@domingohui I didn't check in details how vertical rhythm is handled but it seems that is has changed with your PR. Or is it my 👀  ?

![icons-alignement](https://cloud.githubusercontent.com/assets/103008/25070047/08a3bbea-2293-11e7-9e7e-34c53b3976e6.png)

domingohui commented:
thanks! Should be better this time..

![image](https://cloud.githubusercontent.com/assets/5943822/25077498/76ca0a7e-22f8-11e7-87da-fb13b1ffe854.png)

domingohui commented:
@DirtyF @ashmaroli  Pls feel free to let me know if I should make any further changes. Thanks.

ashmaroli commented:
setting `vertical-align: text-top` locally looked better to **my eyes**. Either ways, no further comments from my side.

pranavcode commented:
I was working on a site and would really like to have other social media icons for minima, are we planning to merge this soon?

domingohui commented:
@ashmaroli I agree. Not all icons look better with ```text-top```. But it probably looks better overall...

domingohui commented:
@ashmaroli I was wondering if we were going to merge this? Or are there any other changes that I should make?

ashmaroli commented:
@domingohui I don't have the authority to comment on that.
/cc @DirtyF

ashawley commented on _includes/social_links/dribbble.html:
```diff
@@ -0,0 +1 @@
+<a href="https://dribbble.com/{{ include.username || site.dribbble_username }}"><svg class="svg-icon" xmlns="http://www.w3.org/2000/svg"><use xlink:href="/assets/icons.svg#dribbble"></use></svg> <span class="username">{{ site.dribbble_username }}</span></a>
```
Each of these social templates has `include.username` which for some reason is being used instead of `site.dribble_username`  when it's probably uninitialized.

ashmaroli replied:
@domingohui Please make this change. `include.username || ` is not required since you have not abstracted the usernames as a parameter for the includes.

Thanks @ashawley :+1:

ashawley replied:
Apparently, this is because the *About* page does the following:

```
You can find the source code for the Jekyll new theme at:
{% include social_links/github.html username="jekyll" %} /
[minima](https://github.com/jekyll/minima)

You can find the source code for Jekyll at
{% include social_links/github.html username="jekyll" %} /
[jekyll](https://github.com/jekyll/jekyll)
```

This is why the social templates were previously separate files I guess.

ashmaroli replied:
This was changed in Jekyll 3.5 AFAIR.. so we should be fine since we're moving to depend on at least v3.5 from the next release onwards

ashawley commented on _includes/footer.html:
```diff
@@ -22,22 +22,23 @@ <h2 class="footer-heading">{{ site.title | escape }}</h2>
+            {% if site.youtube_username %}<li>{% include social_links/youtube.html %}</li>{% endif %}
+            {% if site.googleplus_username %}<li>{% include social_links/googleplus.html %}</li>{% endif %}
+            {% if site.rss %}<li>{% include social_links/rss.html %}</li>{% endif %}
         </ul>
```
Why put every single social site in a separate include?

Why not just make one include called `social.html`, include it from `footer.html` and have it contain all the links in a nicely formatted template with if-conditionals on separate lines?

Just as was done with the combined SVG?

ashmaroli replied:
I like this idea..

ashmaroli commented:
For Minima 3.0, lets group the social usernames under a distinct namespace: `site.social_links`. Whatsay?
```yaml
# _config.yml

social_links  :
  facebook    : johndoe
  twitter     : john-doe
  google_plus : johnnyd
```

DirtyF commented:
@ashmaroli yeah we should stick to a namespace, I would stick to the [existing `jekyll-seo-tag` one ](https://github.com/jekyll/jekyll-seo-tag#usage)`social`.

ashmaroli commented:
> I would stick to the existing `jekyll-seo-tag` one `social`

I'm fine with it as long as it doesn't result in confusion as what subkey is for `Minima` and what subkeys are for `JekyllSEOTag`

domingohui commented:
I notice the social tag contains an array of links. I might have missed something about it. But any idea on how to reference those in an include? So we don't have to duplicate links. If this is possible, do you think this is a good idea?

ashmaroli commented:
@domingohui to avoid breaking sites using the current format **`{{ site.twitter_username }}`**, let's postpone the `social` namespace for a future date.

For now, let's proceed to condense the individual `include` files into a single `include` file.

codingo commented:
I'm using the Minima theme with Github Pages so I don't have local files (such as the footer) to append to. It's not entirely clear to me how this would be done until the namespace solution suggested by @ashmaroli is implemented, could the documentation be expanded to include some information about this?

ashmaroli commented:
> could the documentation be expanded to include some information about this?

@codingo What exactly are you trying to achieve?

codingo commented:
@ashmaroli I'd like to add social links outside of Github/Twitter however it appears this may not yet be a part of minima, or if it is, not yet documented?

ashmaroli commented:
> it appears this may not yet be a part of minima

Correct, the public release version (and hence GitHub Pages) doesn't support social icons outside GitHub n Twitter yet. To achieve that, you'll have to maintain a custom footer and build locally and then push the generated site to GitHub Pages.

codingo commented:
Makes sense. Thank-you!

ashawley commented:
The work of adding social links seems to have stalled here.  I'm going to open a new PR and see if I can get @domingohui's improvements over the line by making the requested changes and resolving another issue I uncovered.

DirtyF commented:
closing this in favor of #141
